src/components/menu/menu.tsx

```diff
@@ -124,28 +99,33 @@ export class Menu {
                     ${this.disabled ? '' : 'menu__trigger-enabled'}
                 `}
                 disabled={this.disabled}
-                onClick={this.onTriggerClick}
             >
                 <span>{this.label}</span>
             </button>
         );
     }
 
-    private handleMdcSelected(event: CustomEvent) {
-        this.select.emit(
-            this.items[
-                parseInt(event.detail.item.dataset.index, 10)
-            ] as ListItem
-        );
-        this.open = false;
-    }
-
-    private handleMdcClosed() {
+    private onClose = () => {
         this.cancel.emit();
         this.open = false;
-    }
+    };
 
     private onTriggerClick = () => {
+        if (this.disabled) {
+            return;
+        }
         this.open = !this.open;
     };
+
+    private onListChange = (event) => {
+        this.items = this.items.map((item: ListItem) => {
+            if (item.value === event.detail.value) {
```

Comment:
I think value has to be a primitive object in order for this to work

Reply:
We should just have to compare the items I guess :thinking: 
```suggestion
            if (item === event.detail) {
```

Reply:
Then I could probably remove most of my changes in the web client :)

Reply:
:smile: